$DragonFly: src/usr.sbin/802_11/wpa_supplicant/driver_wired.c.patch,v 1.1 2007/08/07 11:25:37 sephe Exp $
diff -urp wpa_supplicant-0.5.8/driver_wired.c /usr/src/contrib/wpa_supplicant-0.5.8/driver_wired.c
--- driver_wired.c	2007-05-29 08:41:52.000000000 +0800
+++ driver_wired.c	2007-07-22 11:35:36.000000000 +0800
@@ -18,9 +18,9 @@
 #ifdef __linux__
 #include <netpacket/packet.h>
 #endif /* __linux__ */
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__DragonFly__)
 #include <net/if_dl.h>
-#endif /* __FreeBSD__ */
+#endif /* __FreeBSD__ || __DragonFly__ */
 
 #include "common.h"
 #include "driver.h"
@@ -119,7 +119,7 @@ static int wpa_driver_wired_multi(const 
 	ifr.ifr_hwaddr.sa_family = AF_UNSPEC;
 	os_memcpy(ifr.ifr_hwaddr.sa_data, addr, ETH_ALEN);
 #endif /* __linux__ */
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__DragonFly__)
 	{
 		struct sockaddr_dl *dlp;
 		dlp = (struct sockaddr_dl *) &ifr.ifr_addr;
@@ -131,7 +131,7 @@ static int wpa_driver_wired_multi(const 
 		dlp->sdl_slen = 0;
 		os_memcpy(LLADDR(dlp), addr, ETH_ALEN); 
 	}
-#endif /* __FreeBSD__ */
+#endif /* __FreeBSD__ || __DragonFly__ */
 
 	if (ioctl(s, add ? SIOCADDMULTI : SIOCDELMULTI, (caddr_t) &ifr) < 0) {
 		perror("ioctl[SIOC{ADD/DEL}MULTI]");
